### DISCUSSION TITLE
DNS IP и "перенаправления"

### DISCUSSION THREAD

**Discussion by ammnt** [AdguardTeam/AdGuardHome#3292] [Q&A]
Привет,

мы имеем раздел перенаправлений DNS, файл hosts и предустановленные IP серверов bootstrap. Есть ли возможность сопоставить вручную IP-адреса и домены DNS? Сейчас это сделать нельзя ни через hosts, ни через rewrites, ни удалив bootstrap-адреса полностью🤌🏼

Это нужно за тем, чтобы серверы DoT/DoH/DoQ не нужно было отрезолвливать, если их IP-адреса известны заведомо🙄

Спасибо🖐🏻

**Comment by ameshkov** [comment]
Не очень понятен мне вопрос честно говоря. А что мешает сейчас сопоставить?

**Reply by ammnt** [reply]
@ameshkov, сопоставить ничего не мешает, но IP адреса DNSов все равно будут браться с bootstrap :) Другими словами: предположим, что мы хотим добавить в качестве upstream серверов следующие:
https://cloudflare-dns.com/dns-query
https://dns.google/dns-query
https://dns9.quad9.net/dns-query

Мы заведомо знаем, что у них будут IP соответственно:
1.1.1.1
8.8.8.8
9.9.9.9

Я не вижу смысла обращаться в этом случае к boostrap и тем самым увеличивать время отклика, поэтому хочу просто сопоставить их при помощи раздела rewrites. Но эти записи будут игнорироваться даже в том случае, если я удалю все bootstrap IP, так как там есть предустановленные. Ок?

**Reply by ameshkov** [reply]
Для этого не rewrites надо использовать, а DNS stamps в которых строго заданы IP адреса: https://dnscrypt.info/stamps/

Вот например стамп для гугла с заданным IP-адресом:
`sdns://AgAAAAAAAAAABzguOC44LjgACmRucy5nb29nbGUKL2Rucy1xdWVyeQ`

**Reply by ammnt** [reply]
@ameshkov, но в этом же случае upstream будут работать по протоколу DNSCrypt или всё же DoH?

**Reply by ameshkov** [reply]
По указанному, в DNS stamp можно любой протокол задать.

**Reply by ammnt** [reply]
@ameshkov, хорошо. Спасибо за ответ!
